### DISCUSSION TITLE
What's the secret of nyoom's speed?

### DISCUSSION THREAD

**Discussion by Nias26** [nyoom-engineering/nyoom.nvim#159] [Q&A] [answered]
It's really impressive how fast is nyoom, even at first startup.
I would like to know if it's fennel or the configuration that are optimated and blazingly fast.

**Comment by thebashpotato** [comment, marked as answer]
I'm not a fennel expert, but based off this [fenel langs rational doc](https://fennel-lang.org/rationale) fennel just runs on the lua runtime, which suggests fennel will run at the same speed as lua. I think if you follow a good strategy on your neovim config it will be fast. Most notably using the new [vim.loader.enable() feature](https://neovim.io/doc/user/lua.html#vim.loader.enable()) which uses the byte code complitation cache.  I use it in [my config](https://github.com/thebashpotato/neovim.config) and its really fast. If you need direction on writing your own lua config, how to structure plugins and set everything up for speed and understanding I recommend watching [the neovim from scratch series](https://youtu.be/ctH-a-1eUME?si=KnI2Ko-wycCaqKia). As far as first startup, byte compiled lua is fast, and if you use the lazy.nvim package manager, you can lazy load certain plugins. That is basically how people make fast and efficient neovim setups. As cool as the Nyoom project is, you don't need it or an obscure functional lisp style language like Fennel to have a fast and snappy setup.

**Reply by Nias26** [reply]
Thanks!
